### PR TITLE
World Bounds Fixes

### PR DESCRIPTION
**Pull request by randomPoison**
- Adds a bunch of colliders around the outside of the world the keep the player from leaving the world.
- Adds a new volume type that triggers midnight immediately, and adds these to a bunch of places the player shouldn't be able to get to.

### CODE REVIEW

File: Assets/Scripts/Managers/DayCycleManager.cs [hunk 1]
```diff
@@ -69,8 +69,7 @@ void Update ()
 #if UNITY_EDITOR
 		if ( Input.GetKey( _midnightTrigget ) && _midnightCoroutine == null )
```
Comment (brickman1444) [inline Assets/Scripts/Managers/DayCycleManager.cs]: trigget typo?

File: Assets/Scripts/Managers/DayCycleManager.cs [hunk 2]
```diff
@@ -103,22 +113,27 @@ public static void DeregisterEndOfDayCallback( EndOfDayCallback callback )
...
 		while ( true )
 		{
-			if ( elapsedTime < _blackoutDuration )
+			if ( elapsedTime < overlayTime )
 			{
 				elapsedTime += Time.deltaTime;
 				Color overlayColor = _midnightOverlay.color;
-				overlayColor.a = elapsedTime / _blackoutDuration;
+				overlayColor.a = elapsedTime / overlayTime;
 				_midnightOverlay.color = overlayColor;
```
Comment (brickman1444) [inline Assets/Scripts/Managers/DayCycleManager.cs]: can you replace this with _midnightOverlay.color = midnightOverlay.color.SetAlpha( elapsedTime / overlayTime); ?

File: Assets/Scripts/MidnightVolume.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class MidnightVolume : MonoBehaviour
+{
+	[SerializeField] float _midnightFadeTime = 2.0f;
+
+	void OnTriggerEnter()
+	{
+		DayCycleManager.TriggerMidnight( _midnightFadeTime );
```
Comment (brickman1444) [inline Assets/Scripts/MidnightVolume.cs]: Should probably check that it's the player entering the volume

**Comment by brickman1444** [comment]
Done

File: Assets/Scripts/Managers/DayCycleManager.cs (hunk 1 above)
Comment (randomPoison) [inline Assets/Scripts/Managers/DayCycleManager.cs]: Oh geez, I bet that's been there for a while.

File: Assets/Scripts/Managers/DayCycleManager.cs (hunk 2 above)
Comment (randomPoison) [inline Assets/Scripts/Managers/DayCycleManager.cs]: Done.

File: Assets/Scripts/MidnightVolume.cs (hunk 3 above)
Comment (randomPoison) [inline Assets/Scripts/MidnightVolume.cs]: Done.

**Pull request merged**